### PR TITLE
Added extraVolumes and extraVolumesMounts.

### PR DESCRIPTION
**Pull request by martin-adema**
Closes

Added extraVolumes and extraVolumesMounts values to be able to mount extra volumes.

We use the Secret Store CSI Driver (https://secrets-store-csi-driver.sigs.k8s.io/introduction.html) to mount the OIDC client secret from Azure Keyvault.
To be able to use this driver a volume needs to be added to the deployment with reference to the driver and the SecretProviderClass.

**How was this change implemented?**

Change has been tested locallly with specifying the extraVolumes and extraVolumesMounts values and verifying the resulting deployment definition and tested with no value for extraVolumes and extraVolumesMounts.

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: charts/gitops-server/values.yaml [hunk 1]
```diff
@@ -119,6 +119,8 @@ ingress:
   #  - secretName: chart-example-tls
   #    hosts:
   #      - chart-example.local
+extraVolumes: []
+extraVolumeMounts: []
```
Comment (makkes) [inline charts/gitops-server/values.yaml:123]: I believe it'd be convenient for users to provide examples for both of these settings to understand the syntax.

**Comment by makkes** [comment]
lgtm but please squash the commits before we can merge.

**Review by makkes — APPROVED** [review]

**Review by makkes — CHANGES_REQUESTED** [review]
Sorry by squashing I mean locally squash all commits into one. We want this PR to be one commit.

**Comment by martin-adema** [comment]
Sorry accidently merged from main. Fixed it and squashed the commits.

**Review by makkes — APPROVED** [review]

**Pull request merged**